### PR TITLE
close the socket

### LINKED ISSUE #1361: PcapLiveDevice socket is not close
in file Pcap++/src/PcapLiveDevice.cpp, line 853 and 909 create socket, but never close

### PR DESCRIPTION
**Pull request by tigercosmos**
fix #1361

### CODE REVIEW

**Comment by tigercosmos** [comment]
@seladb  this is ready for review.

File: Pcap++/src/PcapLiveDevice.cpp [hunk 1]
```diff
@@ -859,6 +859,7 @@ void PcapLiveDevice::setDeviceMtu()
 	}
 
 	m_DeviceMtu = ifr.ifr_mtu;
+	::close(socketfd);
```
Comment (seladb) [inline Pcap++/src/PcapLiveDevice.cpp:862]: Maybe we should refactor this code to make sure the socket always closes?
I mean something like this:
```cpp
int socketfd = -1;
try
{
	int socketfd = socket(AF_INET, SOCK_DGRAM, IPPROTO_IP);
	...
}
catch (const std::exception& e)
{
 	//log exception using e.what()
}
if (socketfd != -1)
{
	::close(socketfd);
}
```

File: Pcap++/src/PcapLiveDevice.cpp [hunk 2]
```diff
@@ -914,6 +915,7 @@ void PcapLiveDevice::setDeviceMacAddress()
 	}
 
 	m_MacAddress = MacAddress(ifr.ifr_hwaddr.sa_data[0], ifr.ifr_hwaddr.sa_data[1], ifr.ifr_hwaddr.sa_data[2], ifr.ifr_hwaddr.sa_data[3], ifr.ifr_hwaddr.sa_data[4], ifr.ifr_hwaddr.sa_data[5]);
+	::close(socketfd);
```
Comment (seladb) [inline Pcap++/src/PcapLiveDevice.cpp:918]: ditto

File: Pcap++/src/PcapLiveDevice.cpp (hunk 1 above)
Comment (tigercosmos) [reply Pcap++/src/PcapLiveDevice.cpp]: fixed.

**Review by seladb — APPROVED** [review]

**Pull request merged**